### ISSUE TITLE
Password derivation not working consistently between iOS and Android

### ISSUE THREAD

**Issue by thebalaa** [paritytech/parity-signer#1914]
### On iOS
A password derivation of ///password is accepted

iOS Polkadot Vault Version: 6.1.0

### On Android
A password derivation path of ///password is not accepted, the "Create Derived Key" remains grayed out.

A password derivation path of //polkador///password is accepted on Android

Android Polkadot Vault Version: 6.1.2

**Comment by krodak** [comment]
@Dmitry-Borodin iOS confirms password by first checking for conflicts using Rust API:
https://github.com/paritytech/parity-signer/blob/master/ios/PolkadotVault/Screens/DerivedKey/DerivationPathNameView.swift#L292
and then confirming that password entered is correct is second textfield:
https://github.com/paritytech/parity-signer/blob/master/ios/PolkadotVault/Screens/DerivedKey/DerivationPathNameView.swift#L299
which I think is correct behaviour, cc @prybalko 

@thebalaa just to confirm - did you entered password confirmation in second textfield on both platforms, could you attach example screenshot / gif / video? Thanks in advance 🙏🏻

**Comment by thebalaa** [comment]
Yes, this happens when attempting to enter password in second text field (password confirmation) on both platforms.

**Comment by krodak** [comment]
![2023-07-03 08 27 04](https://github.com/paritytech/parity-signer/assets/1955364/7a7df01f-8375-40cb-b9ca-d74a4d8f3308)

@Dmitry-Borodin flow on iOS

**Comment by Dmitry-Borodin** [comment]
Added support for empty path only password derivation on android - I thought it's not a valid path.

btw @krodak I use rust check only to check for collision - it's in DerivationCreateViewModel.kt:90 - the rest should work locally. But it's fast enough anyway, I think doing rust checks always is not an issue.
